docs/source/user-guide/expressions.md

```diff
@@ -304,6 +304,16 @@ select log(-1), log(0), sqrt(-1);
 | rollup(exprs)                                                     | Creates a grouping set for rollup sets.                                                 |
 | sum(expr)                                                         | Сalculates the sum of `expr`.                                                           |
 
+## Aggregate Function Builder
+
+Another builder expression that ends with `build()`, it is useful if the functions has multiple optional arguments
+
+See datafusion-examples/examples/udaf_expr.rs for example usage.
+
+| Syntax                                                 | Equivalent to                       |
+| ------------------------------------------------------ | ----------------------------------- |
+| first_value_builder(expr).order_by(vec![expr]).build() | first_value(expr, Some(vec![expr])) |
```

Comment:
I think we can add non-trivial examples in doc, but not all of them.

Reply:
Yes I think adding examples in doc comments is likely the best for this particular API as people will be looking at the code already if they want to programatically create Exprs